packages/chat-stateful-client/src/ChatClientState.ts

```diff
@@ -14,6 +14,12 @@ export type ChatClientState = {
    * Object with {@Link ChatThreadClientState} fields, keyed by {@Link ChatThreadClientState.threadId}.
    */
   threads: { [key: string]: ChatThreadClientState };
+  /**
+   * Stores the latest error for each API method.
+   *
+   * See documentation of {@Link ChatErrors} for details.
+   */
+  latestErrors: ChatErrors;
```

Comment:
Sorry, might be a stupid question but would we have a selector for this in the future to surface errors to the UI? For example, if we failed to change the topic of the chat. The error would be available in latestErrors['ChatThreadClient.updateTopic']? Would the user of the selector would need to know the stateful function name?

Reply:
Yes and no.

There will be some generic selectors which just show _any_ error from the object (e.g, show the latest 3 errors in an `ErrorBar`). Those will not need to know the exact values.

There will be others that need to pick out a particular error target (e.g., disable the `SendBox` when we failed to send message because the backend throttled us). Those need to know the operation that failed in _some_ way. I think directly using the method names is best. Otherwise, we'd end up creating another enum that somehow maps to the methods. The bindings will also get the type information, so you can auto-complete to find what property names are valid in `ChatErrors`.